block-storage/src/main/scala/coop/rchain/blockstorage/KeyValueBlockStore.scala

```diff
@@ -0,0 +1,161 @@
+package coop.rchain.blockstorage
+
+import java.nio.ByteBuffer
+
+import cats.effect.Sync
+import cats.syntax.all._
+import com.google.protobuf.{ByteString, CodedInputStream}
+import coop.rchain.casper.PrettyPrinter
+import coop.rchain.casper.protocol._
+import coop.rchain.models.BlockHash.BlockHash
+import coop.rchain.shared.ByteStringOps.RichByteString
+import coop.rchain.shared.ByteVectorOps.RichByteVector
+import coop.rchain.shared.Compression
+import coop.rchain.shared.syntax._
+import coop.rchain.store.{KeyValueStore, KeyValueStoreManager}
+import net.jpountz.lz4.{LZ4CompressorWithLength, LZ4DecompressorWithLength}
+import scodec.bits.ByteVector
+
+class KeyValueBlockStore[F[_]: Sync](
+    private val store: KeyValueStore[F],
+    private val storeApprovedBlock: KeyValueStore[F]
```

Comment:
It's a bit confusing to see here approved block, as you've mentioned that approved block store is now part of DAG database

Reply:
Yes, for now I didn't want to touch existing BlockStore interface because I don't think they are needed at all. They are just two instances of `KeyValueTypedStore`.

To be able to do this, interface for searching should be defined separately, something like this.
```scala
trait Searchable[F[_], A, B] {
  def find(term: A): F[Seq[B]]
}
```
We can even have defaults for `KeyValueStore` e.g. `Searchable[F, Hex[String], Array[Byte]]` that can search keys in any store as hex text.